reform-db/cmd_init.go

```diff
@@ -0,0 +1,177 @@
+		using[i] = fmt.Sprintf("key_column_usage.%s = table_constraints.%s", u, u)
+	}
+	q := fmt.Sprintf(
+		`SELECT column_name, ordinal_position FROM information_schema.key_column_usage
+			INNER JOIN information_schema.table_constraints ON %s
+			WHERE key_column_usage.table_catalog = %s AND
+				key_column_usage.table_schema = %s AND
+				key_column_usage.table_name = %s AND
+				constraint_type = 'PRIMARY KEY'
+			ORDER BY ordinal_position DESC`,
+		strings.Join(using, " AND "), db.Placeholder(1), db.Placeholder(2), db.Placeholder(3),
+	)
+	row := db.QueryRow(q, catalog, schema, tableName)
+	var key keyColumnUsage
+	if err := row.Scan(key.Pointers()...); err != nil {
+		if err == reform.ErrNoRows {
+			return nil
+		}
+		logger.Fatalf("%s", err)
+	}
+	if key.OrdinalPosition > 1 {
+		logger.Fatalf("Expected single column primary key, got %d", key.OrdinalPosition)
+	}
+	return &key
+}
+
+// initModelsInformationSchema returns structs from database with information_schema.
+func initModelsInformationSchema(db *reform.DB, tablesTail string, typeFunc typeFunc) (structs []StructData) {
+	tables, err := db.SelectAllFrom(tableView, tablesTail)
+	if err != nil {
+		logger.Fatalf("%s", err)
+	}
+
+	imports := make(map[string]struct{})
+	for _, t := range tables {
+		table := t.(*table)
+		str := parse.StructInfo{Type: toCamelCase(table.TableName), SQLName: table.TableName}
+		var comments []string
+
+		key := getPrimaryKeyColumn(db, table.TableCatalog, table.TableSchema, table.TableName)
+
+		tail := fmt.Sprintf(
+			`WHERE table_catalog = %s AND table_schema = %s AND table_name = %s ORDER BY ordinal_position`,
+			db.Placeholder(1), db.Placeholder(2), db.Placeholder(3),
+		)
+		columns, err := db.SelectAllFrom(columnView, tail, table.TableCatalog, table.TableSchema, table.TableName)
+		if err != nil {
+			logger.Fatalf("%s", err)
+		}
+		for i, c := range columns {
+			column := c.(*column)
+			typ, pack, comment := typeFunc(column.Type, bool(column.IsNullable))
+			if pack != "" {
+				imports[pack] = struct{}{}
+			}
+			comments = append(comments, comment)
+			str.Fields = append(str.Fields, parse.FieldInfo{
+				Name:   toCamelCase(column.Name),
+				Type:   typ,
+				Column: column.Name,
+			})
+
+			if key != nil && key.ColumnName == column.Name {
+				str.PKFieldIndex = i
+			}
+		}
+
+		structs = append(structs, StructData{
+			Imports:       imports,
+			StructInfo:    str,
+			FieldComments: comments,
+		})
+	}
+
+	return
+}
+
+// cmdInit implements init command.
+func cmdInit(db *reform.DB, dir string) {
+	var structs []StructData
+	switch db.Dialect {
+	case postgresql.Dialect:
+		structs = initModelsInformationSchema(db, `WHERE table_schema = current_schema()`, goTypePostgres)
+	case mysql.Dialect:
+		structs = initModelsInformationSchema(db, `WHERE table_schema = DATABASE()`, goTypeMySQL)
+	case sqlite3.Dialect:
+		structs = initModelsSQLite3(db)
+	case mssql.Dialect:
+		structs = initModelsInformationSchema(db, ``, goTypeMSSQL)
```

Comment:
Try this:
```sql
select * from information_schema.tables where table_catalog = db_name()
```

![1](https://cloud.githubusercontent.com/assets/4426185/23590886/04ba54ae-01f9-11e7-93d6-fdbc24efc750.png)